cpp/src/arrow/dataset/file_base.h

```diff
@@ -108,6 +115,9 @@ class ARROW_DS_EXPORT FileSource : public util::EqualityComparable<FileSource> {
   /// \brief Get a RandomAccessFile which views this file source
   Result<std::shared_ptr<io::RandomAccessFile>> Open() const;
 
+  /// \brief Get the size (in bytes) of the file or buffer
```

Comment:
Hmmm...the same question exists for "encoded size?" or "decoded size?".  I believe the answer is "uncompressed and decoded".  The goal is to limit how much data is loaded after a scanner is paused.  I will update this comment.

Reply:
Ah, I misunderstood and thought we were talking about a different size.  This size could pose a problem for compressed streams.  It should be the compressed size since this is referring to the size of the file.  I'll have to think over the implications (or test) of compressed streams a bit more.